lalrpop/src/normalize/lower/mod.rs

```diff
@@ -78,16 +78,24 @@ impl<'s> LowerState<'s> {
                                                     let pattern = Pattern {
                                                         span: span,
                                                         kind: PatternKind::Tuple(vec![
-                                        Pattern {
-                                            span: span,
-                                            kind: PatternKind::Usize(index),
-                                        },
-                                        Pattern {
-                                            span: span,
-                                            kind: PatternKind::Choose(input_str.clone())
-                                        }
-                                        ]),
+                                                                    Pattern {
+                                                                        span: span,
+                                                                        kind: PatternKind::Usize(index),
+                                                                    },
+                                                                    Pattern {
+                                                                        span: span,
+                                                                        kind: PatternKind::Choose(input_str.clone())
+                                                                    }
+                                                                    ]),
                                                     };
+
+                                                    // FIXME: This should be cleaner
```

Comment:
if we replaced `literals` with a `Vec` of structs, or something like that, then it would be nicer